src/beamlime/applications/_nexus_helpers.py

```diff
@@ -1,155 +1,365 @@
-        store[path] = dict(
-            children=[],
-            name=_node_name(parent),
+    """Merges message into the associated nexus group.
+
+    If there are multiple paths that match the message,
+    all paths will be used.
+    In principle, there should only be one path that matches the message.
+
+    But the nexus template does not guarantee that there is a unique matching module.
+    There are still some nexus templates that contain multi-matching
+    module place holders (i.e. same ``source_name`` in ev44 module).
+
+    In this case, it is more safe to use all paths
+    so that no data is lost in the data reduction,
+    even though it means unnecessary memory usage
+    and slower in the grouping/binning process
+    (``N`` times slower where ``N`` is number of duplicating modules).
+
+    For example, if we store all data in to every detector data bank,
+    then irrelevant data points will not affect the data reduction
+    once the data is grouped by the pixel ids.
+    However, if we choose to store the data in only one of the detector data banks,
+    then the data of the rest of detectors will be lost.
+    """
+    if kind == "ev44":
+        _merge_message_into_data_module_store(
+            data_module_store=data_module_store,
+            structure=structure,
+            data_piece=data_piece,
+            path_matching_func=find_ev44_matching_paths,
+            data_field_initialize_func=_initialize_ev44,
+            merge_func=_merge_ev44,
         )
-        if 'attributes' in parent:
-            store[path]['attributes'] = parent['attributes']
-        if len(parent['children']) > 1:
-            raise ValueError("Multiple modules found in the same data group.")
-    kind, content = message
-    if kind == 'ev44':
-        _merge_ev44(store[path], content)
-    elif kind == 'f144':
-        _merge_f144(store[path], content)
-    elif kind == 'tdct':
-        _merge_tdct(store[path], content)
     else:
         raise NotImplementedError
 
 
-def combine_store_and_structure(
-    store: Mapping[Tuple[Optional[str], ...], Mapping], structure: Mapping
-):
-    '''Creates a new nexus structure, replacing the stream modules
+def combine_data_module_store_and_structure(
+    data_module_store: DataModuleStore, structure: Mapping
+) -> Mapping:
+    """Creates a new nexus structure, replacing the stream modules
     with the datasets in `store`, while avoiding
-    to copy data from `structure` if unessecary'''
-    if len(store) == 0:
+    to copy data from `structure` if unnecessary"""
+    if len(data_module_store) == 0:
         return structure
-    if (None,) in store:
-        return store[(None,)]
+    if (None,) in data_module_store:
+        return data_module_store[(None,)]
 
-    new = structure.copy()
-    if 'children' in structure:
+    new = {**structure}  # Faster than `dict()`
+    # and avoid static type-error of `copy`
+    # since `structure` is `Mapping`
```

Comment:
(minor) I think `{**structure}` is absolutely fine as a replacement for `.copy`. However, maybe we should just change the argument type of `structure` to `dict` anyway. It is a dict, after all. It's up to you!

In any case, the comments explaining this are not necessary.